### ISSUE TITLE
Add UnnecessaryFilter

### ISSUE THREAD

**Issue by VovaStelmashchuk** [detekt/detekt#3340]
## Expected Behavior of the rule
```kotlin
val x = listOf(1, 2, 3)
      .filter { it > 1 }
      .size
```
No erros

## Context
Everethink should be simple.
Add create a PR only with tests, I haven't idea how to create implementation at the moment.

**Comment by BraisGabin** [comment]
I like the idea. We could check it for `Sequence`s too. And thanks for implement the tests!

But I must say that I didn't understand your PR until I read the tests you provided. Could you elaborate a bit more this issue? This way other person can pick it easily and implement it.

**Comment by VovaStelmashchuk** [comment]
In same case we need to calculate count of items in collection by predicate. Sometimes people use `filter{}.size` or `filter {}.count()`. The correct way to use kotlin function count {}. Because filter create a new instanse of collection.

**Comment by chao2zhang** [comment]
Close via #3341
